### PR TITLE
enhancement(external docs): Add status field to Remap function docs

### PR DESCRIPTION
**Pull request by lucperkins**
This PR adds a `status` field to Remap functions that enables us to mark functions as stable or experimental (with stable as the default). This field is not currently necessary, as all functions are stable, but this will likely change in the future as new approaches are tried out.

### CODE REVIEW

File: docs/reference/remap.cue [hunk 1]
```diff
@@ -47,13 +47,15 @@ remap: {
 				source: string
 				output: #Fields
 			}
+			#Status: *"stable" | "experimental"
```
Comment (binarylogic) [inline docs/reference/remap.cue:50]:
```suggestion
			#Status: *"stable" | "beta"
```

Thoughts on that? It's inline with our component statuses.

Comment (binarylogic) [reply docs/reference/remap.cue:50]: Actually, we already have `#DevelopmentStatus`

File: docs/reference/remap.cue [hunk 2]
```diff
@@ -47,13 +47,15 @@ remap: {
...
 
 			arguments: [...#Argument] // Allow for empty list
 			return: [#RemapReturnTypes, ...#RemapReturnTypes]
 			category:    "coerce" | "numeric" | "object" | "parse" | "text" | "hash" | "event" | "networking"
 			description: string
 			examples?: [#RemapExample, ...#RemapExample]
-			name: Name
+			name:   Name
+			status: #Status
```
Comment (binarylogic) [inline docs/reference/remap.cue:58]:
```suggestion
			status: #DevelopmentStatus
```

**Comment by lucperkins** [comment]
Several weeks after opening this, it's not yet clear that we plan on having beta functions. I'll close this for now and prepare to re-open if we go down that road.

**Pull request closed without merging**